packages/cli-platform-ios/src/commands/buildIOS/index.ts

```diff
@@ -0,0 +1,239 @@
+
+  logger.info(
+    `Found Xcode ${
+      xcodeProject.isWorkspace ? 'workspace' : 'project'
+    } "${chalk.bold(xcodeProject.name)}"`,
+  );
+
+  const extendedArgs = {
+    ...args,
+    packager: false,
+  };
+
+  // // No need to load all available devices
+  if (!args.device && !args.udid) {
+    /**
+     * If provided simulator does not exist, try simulators in following order
+     * - iPhone 14
+     * - iPhone 13
+     * - iPhone 12
+     * - iPhone 11
+     */
+    const fallbackSimulators = [
+      'iPhone 14',
+      'iPhone 13',
+      'iPhone 12',
+      'iPhone 11',
+    ];
+
+    const selectedSimulator = getDestinationSimulator(args, fallbackSimulators);
+
+    return buildProject(
+      xcodeProject,
+      selectedSimulator.udid,
+      scheme,
+      extendedArgs,
+    );
+  }
+
+  if (args.device && args.udid) {
+    return logger.error(
+      'The `device` and `udid` options are mutually exclusive.',
+    );
+  }
+
+  const devices = getDevices();
+
+  if (args.udid) {
+    const device = devices.find((d) => d.udid === args.udid);
+    if (!device) {
+      return logger.error(
+        `Could not find a device with udid: "${chalk.bold(
+          args.udid,
+        )}". ${printFoundDevices(devices)}`,
+      );
+    }
+
+    return buildProject(xcodeProject, device.udid, scheme, extendedArgs);
+  } else {
+    const physicalDevices = devices.filter((d) => d.type !== 'simulator');
+    const device = matchingDevice(physicalDevices, args.device);
+    if (device) {
+      return buildProject(xcodeProject, device.udid, scheme, extendedArgs);
+    }
+  }
+}
+
+function matchingDevice(
+  devices: Array<Device>,
+  deviceName: string | true | undefined,
+) {
+  if (deviceName === true) {
+    const firstIOSDevice = devices.find((d) => d.type === 'device')!;
+    if (firstIOSDevice) {
+      logger.info(
+        `Using first available device named "${chalk.bold(
+          firstIOSDevice.name,
+        )}" due to lack of name supplied.`,
+      );
+      return firstIOSDevice;
+    } else {
+      logger.error('No iOS devices connected.');
+      return undefined;
+    }
+  }
+  const deviceByName = devices.find(
+    (device) =>
+      device.name === deviceName || formattedDeviceName(device) === deviceName,
+  );
+  if (!deviceByName) {
+    logger.error(
+      `Could not find a device named: "${chalk.bold(
+        String(deviceName),
+      )}". ${printFoundDevices(devices)}`,
+    );
+  }
+  return deviceByName;
+}
+
+function formattedDeviceName(simulator: Device) {
+  return simulator.version
+    ? `${simulator.name} (${simulator.version})`
+    : simulator.name;
+}
+
+function printFoundDevices(devices: Array<Device>) {
+  return [
+    'Available devices:',
+    ...devices.map((device) => `  - ${device.name} (${device.udid})`),
+  ].join('\n');
+}
+
+export default {
+  name: 'build-ios',
+  description: 'builds your app on iOS simulator',
+  func: buildIOS,
+  examples: [
```

Comment:
Examples are mentioning `run-ios` command... 😉 
Also, `device`/`simulator` options should be gone I think?